python/sdk/merlin/batch/source.py

```diff
@@ -106,7 +107,101 @@ def to_client_bq_source(self) -> client.PredictionJobConfigBigquerySource:
                 opts[k] = v
 
         return client.PredictionJobConfigBigquerySource(
+            table=self._table, features=list(self._features), options=opts
+        )
+
+
+class MaxComputeSource(Source):
+    """
+    Source contract for MaxCompute to create prediction job
+    """
+
+    def __init__(
+        self,
+        table: str,
+        features: Iterable[str],
+        endpoint: str,
+        options: MutableMapping[str, str] = None,
+    ):
+        """
+
+        :param table: table id if the source in format of `alicloud-project.schema.table_name`
+        :param features: list of features to be used for prediction, it has to match the column name in the source table.
+        :param endpoint: MaxCompute endpoint
+        :param options: additional option to configure source.
+        """
+        self._table = table
+        self._features = features
+        self._options = options
+        self._endpoint = endpoint
+
+    @property
+    def table(self) -> str:
+        return self._table
+
+    @table.setter
+    def table(self, table):
+        self._table = table
+
+    @property
+    def features(self) -> Iterable[str]:
+        return self._features
+
+    @features.setter
+    def features(self, features):
+        self._features = features
+
+    @property
+    def options(self) -> Optional[MutableMapping[str, str]]:
+        return self._options
+
+    @options.setter
+    def options(self, options):
+        self._options = options
+
+    def _validate(self):
+        if not self._valid_types():
+            raise ValueError("invalid input type")
+        if not mc_valid_table_id(self.table):
+            raise ValueError(f"invalid table: {self.table}")
+        if not mc_valid_columns(self.features):
+            raise ValueError(f"invalid features column: {self.features}")
+
+    def _valid_types(self) -> bool:
+        if not isinstance(self._table, str):
+            return False
+        if not isinstance(self._features, list):
+            return False
+        if self._options is not None and not isinstance(self._options, MutableMapping):
+            return False
+        if not isinstance(self._endpoint, str):
+            return False
+        for feature in self._features:
+            if not isinstance(feature, str):
+                return False
```

Comment:
Ahh, you can just join this check into the one above like:
```python
if not isinstance(self._features, list) or not all([isinstance(feature, str) for feature in self._features]):
```
but I don't have a preference really 😅

Reply:
Thanks, fixed